src/test/pybind/test_cephfs.py

```diff
@@ -1624,3 +1627,54 @@ def test_rmtree_on_a_very_very_deep_tree(self, testdir):
         # occur.
         cephfs.rmtree('dir1', should_cancel, suppress_errors=False)
         assert_raises(libcephfs.ObjectNotFound, cephfs.stat, 'dir1')
+
+
+class TestUnlinkAt:
+
+    def test_unlinkat_regfile(self, testdir):
+        regfilename = 'file1'
+
+        fd = cephfs.open(regfilename, 'w', 0o755)
+        cephfs.write(fd, b"abcd", 0)
+        cephfs.close(fd)
+
+        fd = cephfs.open('/', os.O_RDONLY | os.O_DIRECTORY, 0o755)
+        cephfs.unlinkat(fd, './' + regfilename, 0)
+        cephfs.close(fd)
+
+    def test_unlinkat_dir(self, testdir):
+        dirname = 'dir1'
+
+        cephfs.mkdir(dirname, 0o755)
+        fd = cephfs.open('/', os.O_RDONLY | os.O_DIRECTORY, 0o755)
```

Comment:
I think it would be more useful to test with the fd for `dirname` rather than `/`.

You should also test that an absolute path like:

unlinkat(dirfd, "/path/to/file"...)

will always unlink the absolute path "/path/to/file" and not "<dirfd>/jpath/to/file".

Reply:
Added more tests to test these cases.